tool/matomo/classes/tool/tool.php

```diff
@@ -41,6 +41,7 @@ public function get_tracking_code() {
         $template = new \stdClass();
         $template->siteid = $settings['siteid'];
         $template->siteurl = $settings['siteurl'];
+        $template->piwikurl = isset($settings['piwikurl'])? $settings['piwikurl']: '';
```

Comment:
I would change this so it defaults to siteurl and remove all the 'if' login inside the template